sycl/include/sycl/kernel_bundle.hpp

```diff
@@ -236,6 +241,21 @@ class kernel_bundle : public detail::kernel_bundle_plain,
     return kernel_bundle_plain::has_kernel(KernelID, Dev);
   }
 
+  /// \returns true only if the kernel bundle contains the kernel identified by
+  /// KernelName.
+  template <typename KernelName> bool has_kernel() const noexcept {
+    using KI = sycl::detail::KernelInfo<KernelName>;
+    return has_kernel_impl(KI::getName());
```

Comment:
Would the following be possible?
```suggestion
    return has_kernel(get_kernel_id<KernelName>());
```
And similar for the function below. That way we do not need the new _impl members as well as new symbols.

Reply:
This didn't work originally but tried it now and just needed to add a forward declaration for it an it works. I'll switch it over to this to avoid duplication.